### DISCUSSION TITLE
Keystone 5 to Keystone 6 migration guide

### DISCUSSION THREAD

**Discussion by pahaz** [keystonejs/keystone#8490] [Technical Discussion]
Hi. We have a `Keystone 5`-based project with more than 100 000 lines of code. 
https://github.com/open-condo-software/condo

I want to leave here some set of problems and solutions encountered along the migration way.

Maybe someone also can leave their experience here.

**What do you need to know?**

 - Keystone6 doesn't have Graphql API compatibility with Keystone5
 - If you are familiar with Keystone5 source code you probably won't be able to find familiar abstractions under the hood of Keystone6. We can say that internally it's a completely different framework. 

**Our journey**

 - We decided to write a light abstraction level between ks5 and ks6 to continue developing new features without stopping
 - We will use JavaScript for our back-end logic despite the fact that ks6 is TypeScript based. We want to make our code work with Keystone 6 and then think about switching to TypeScript.

**Questions** 

Does someone also want to share their migration experience? 
Maybe the keystone maintainers want to provide some migration guide?

**Comment by gautamsi** [comment]
I had to write small migration code which would use graphql to migrate the data, I had to use `oldId` field for mapping all the existing unique id. This is still tedious but should work with minimal effort.

**Reply by pahaz** [reply]
We already use UIDs as our primary keys everywhere. And I think we continue to use it. The main challenge for us at the moment is the Hook and Access API changes. We don't want to rewrite all our access and hooks logic
